ext/openssl/openssl.c

```diff
@@ -6593,12 +6611,15 @@ static int php_openssl_cipher_init(const EVP_CIPHER *cipher_type,
 	if (php_openssl_validate_iv(piv, piv_len, max_iv_len, free_iv, cipher_ctx, mode) == FAILURE) {
 		return FAILURE;
 	}
-	if (mode->is_single_run_aead && enc) {
+	if (mode->should_set_tag_length) {
+		/* Explicitly set the tag length even when decrypting,
```

Comment:
Isn't this just for OCB? Shouldn't CCM be done only for encryption?

Reply:
It's only *necessary* for OCB, but it's also allowed for CCM. It seems cleaner not to make a distinction between them.

Reply:
Ah yeah right it would actually fail a test with a shorter tag. All looks good then.

Reply:
Ah ok. It would actually fail a test with a shorter tag. All looks good then.